src/Tools/SchemaTool.php

```diff
@@ -735,7 +736,7 @@ private function gatherRelationJoinColumns(
             $theJoinTable->addUniqueIndex($unique['columns'], is_numeric($indexName) ? null : $indexName);
         }
 
-        $compositeName = $theJoinTable->getName() . '.' . implode('', $localColumns);
+        $compositeName = ($theJoinTable instanceof AbstractNamedObject ? $theJoinTable->getObjectName()->toString() : $theJoinTable->getName()) . '.' . implode('', $localColumns);
```

Comment:
Can you please introduce a helper function for this ternary expression, similar to what you did in `DatabaseDriver`?